pkg/pkg/local.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package pkg
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ksonnet/ksonnet/pkg/parts"
+
+	"github.com/ksonnet/ksonnet/pkg/app"
+	"github.com/ksonnet/ksonnet/pkg/prototype"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+// Local is a package based on vendored contents.
+type Local struct {
+	a            app.App
+	name         string
+	registryName string
+	config       *parts.Spec
+}
+
+var _ Package = (*Local)(nil)
+
+// NewLocal creates an instance of Local.
+func NewLocal(a app.App, name, registryName string) (*Local, error) {
+	partsPath := filepath.Join(a.VendorPath(), registryName, name, "parts.yaml")
```

Comment:
Is ownership of the cache going to transition from `Registry` to `PackageManager`? Given the work I'm currently doing, the less components touching the cache directly, the better.

Reply:
Wherever it lands, I'd like to see an interface centered around `io.Reader` / `io.Writer`, and less dependence on file path manipulation and `afero` dispersed across the code base.